extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -1247,8 +1247,13 @@ public AuthorizationCodeTokens apply(AuthorizationCodeTokens tokens) {
     private Uni<AuthorizationCodeTokens> getCodeFlowTokensUni(RoutingContext context, TenantConfigContext configContext,
             String code, String codeVerifier) {
 
-        // 'redirect_uri': typically it must match the 'redirect_uri' query parameter which was used during the code request.
+        // 'redirect_uri': it must match the 'redirect_uri' query parameter which was used during the code request.
         String redirectPath = getRedirectPath(configContext.oidcConfig, context);
+        if (configContext.oidcConfig.authentication.redirectPath.isPresent()
```

Comment:
@gastaldi @pedroigor This is the only hardening code that I've added to enforce a precise [ath match at the Quarkus level. For example, if the configured redirect path is `/a/b/c` bbut the current request attempting to complete the code flow is `/a` then it means, assuming the best and wide-spread practice that a registered application in provider X has configured `www.my-provider-app/a/b/c` then this redirect may not be coming from that provider . We haven't have to deal it with it before because the providers will enforce such checks. For example, I linked to Google docs and it can easily be confirmed with all providers here except Strava that a mismatch will cause an eventual failure, but, it costs a remote call, and we rely on providers to do the right thing, so this simple check will make the initial check done early at the Quarkus level which is important for the Strava case.

Reply:
I've found it difficult to test this specific check though as it is difficult to verify the test failed exactly because of this check,  I've spent quite some time today on it

Reply:
@gastaldi Hi George, I've now added a test for this condition by modifying one of the existing CodeFlowTest integration tests, not perfect but is probably closest to what can be done without having to created a dedicated integration module and check logs. So I'm happy enough now.